### PR TITLE
New post fixes

### PR DESCRIPTION
**Pull request by ZeRoGerc**
Saving state and rotation

### CODE REVIEW

File: app/src/main/java/ly/loud/loudly/ui/MainActivity.java [hunk 1]
```diff
@@ -215,14 +228,56 @@ public void startFragment(@NonNull Fragment fragment) {
...
+    }
+
+    /**
+     * Run before back button super method to check if we need to hide post layout from the screen
+     * or just change state of bottom sheet.
+     */
+    private void tryHidePostCreateLayoutBasedOnBottomSheet() {
+        if (bottomSheetBehavior.getState() == STATE_COLLAPSED) {
+            newPostFragmentView.setVisibility(GONE);
+            background.setOpacity(0);
```
Comment (ZeRoGerc) [inline app/src/main/java/ly/loud/loudly/ui/MainActivity.java]: olive

File: app/src/main/java/ly/loud/loudly/ui/adapters/AttachmentAdapter.java [hunk 2]
```diff
@@ -10,21 +10,24 @@
...
 import ly.loud.loudly.base.plain.PlainImage;
+import solid.collections.SolidList;
+
+import static ly.loud.loudly.util.ListUtils.asSolidList;
 
 public class AttachmentAdapter extends RecyclerView.Adapter<AttachmentAdapter.AttachmentHolder>{
 
     @NonNull
-    private List<Attachment> attachmentList;
+    private ArrayList<Attachment> attachmentList;
```
Comment (ZeRoGerc) [inline app/src/main/java/ly/loud/loudly/ui/adapters/AttachmentAdapter.java]: realization not matters

File: app/src/main/java/ly/loud/loudly/ui/new_post/NetworkChooseAdapter.java [hunk 3]
```diff
@@ -64,29 +80,30 @@ public void setOnItemStateChangeListener(@NonNull OnItemStateChangeListener onIt
         @NonNull
         TextView textView;
 
+        /**
+         * Id of currently bound network.
+         */
+        int currentId = 0;
```
Comment (ZeRoGerc) [inline app/src/main/java/ly/loud/loudly/ui/new_post/NetworkChooseAdapter.java]: incapsulation

File: app/src/main/java/ly/loud/loudly/ui/new_post/NetworkChooseAdapter.java [hunk 4]
```diff
@@ -64,29 +80,30 @@ public void setOnItemStateChangeListener(@NonNull OnItemStateChangeListener onIt
...
             super(LayoutInflater.from(context).inflate(R.layout.network_choose_list_item, parent, false));
             ButterKnife.bind(this, itemView);
         }
 
-        public void refresh(@NonNull NetworkContract models) {
-            textView.setText(models.getFullName());
+        public void refresh(@NonNull NetworkContract model) {
+            currentId = model.getId();
+            textView.setText(model.getFullName());
+            if (selected[currentId]) {
```
Comment (ZeRoGerc) [inline app/src/main/java/ly/loud/loudly/ui/new_post/NetworkChooseAdapter.java]: olive

File: app/src/main/java/ly/loud/loudly/ui/views/TextPlusAttachmentsView.java [hunk 5]
```diff
@@ -104,7 +98,34 @@ public String getText() {
...
+        return adapter.getAttachmentList();
+
+    }
 
+    @Override
+    @NonNull
+    public Parcelable onSaveInstanceState () {
+        Bundle state = new Bundle();
+        state.putParcelable(SUPER_STATE, super.onSaveInstanceState());
+        state.putParcelableArrayList(ATTACHMENTS, new ArrayList<>(adapter.getAttachmentList()));
```
Comment (ZeRoGerc) [inline app/src/main/java/ly/loud/loudly/ui/views/TextPlusAttachmentsView.java]: asArrayList

**Comment by danilkolikov** [comment]
Good, but with some issues

**Pull request merged**